grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -526,6 +526,47 @@ class ProductAvailabilityService {
         return quantityMap
     }
 
+    /**
+     * Calculate quantities based on expiration filter.
+     * @param productAvailabilityRecords
+     * @param expirationFilter
+     * @return
+     * Return map with:
+     * expiredQuantityAvailableToPromise - quantity available to promise that is expired based on expiration filter
+     * finalQuantityAvailableToPromise - quantity available to promise after applying expiration filter
+     * quantityOnHand - total quantity on hand (not affected by expiration filter)
+     * quantityAvailableToPromise - total quantity available to promise (not affected by expiration filter)
+     */
+    private static Map<String, Integer> calculateQuantitiesByExpirationFilter(List<ProductAvailability> productAvailabilityRecords, ExpirationFilter expirationFilter) {
+        boolean removeExpiredStock = expirationFilter == ExpirationFilter.SUBTRACT_EXPIRED_STOCK
+        Date today = new Date()
+        Date maxDate = removeExpiredStock ? today : today + expirationFilter.days
+        // Quantity available to promise before applying expiration filter
+        Integer quantityAvailableToPromise = productAvailabilityRecords.sum { it.quantityAvailableToPromise ?: 0 }
+        // Final quantity available to promise - quantity available to promise subtracted by expired stock based on expiration filter
+        Integer finalQuantityAvailableToPromise = productAvailabilityRecords.sum { ProductAvailability paRecord ->
+            // We can return immedietaly if we do not want to subtract expired stock
+            if (expirationFilter == ExpirationFilter.DO_NOT_SUBTRACT_EXPIRED_STOCK) {
```

Comment:
why do we need both `SUBTRACT_EXPIRED_STOCK` and `DO_NOT_SUBTRACT_EXPIRED_STOCK`? Can't we just check `if expirationFilter != ExpirationFilter.SUBTRACT_EXPIRED_STOCK`?